src/measure_theory/interval_integral.lean

```diff
@@ -1345,4 +1345,128 @@ theorem integral_deriv_eq_sub (hderiv : ∀ x ∈ interval a b, differentiable_a
 integral_eq_sub_of_has_deriv_at (λ x hx, (hderiv x hx).has_deriv_at) hcont'
   (measurable_deriv f).ae_measurable
 
+/-!
+### Integration by parts
+ -/
+
+lemma self_mem_ae_restrict {s : set ℝ} (hs : is_measurable s):
+  s ∈ (measure.restrict measure_space.volume s).ae :=
+begin
+  rw ae_restrict_eq hs,
+  simp only [exists_prop, filter.mem_principal_sets, filter.mem_inf_sets],
+  exact ⟨univ, filter.univ_mem_sets, s, by simp⟩,
+end
+
+lemma integral_congr_Ioc {f g : ℝ → ℝ} (h : ∀ x ∈ Ioc a b, f x = g x) :
+  ∫ x in Ioc a b, f x = ∫ x in Ioc a b, g x :=
+integral_congr_ae (filter.eventually_of_mem (self_mem_ae_restrict is_measurable_Ioc) h)
```

Comment:
Can you delete this lemma and use `measure_theory.set_integral_congr` instead?

Reply:
That I haven't been able to do... If it is not direct to prove, then shouldn't something like this be in the library?

Reply:
Funny, the reason you haven't been able to do it is that the lemma was only added to the library four days ago! #5647

If you merge master, you should get access to the lemma.  The usage is nearly the same -- you'll write 
```
set_integral_congr is_measurable_Ioc (λ x hx, h x (mem_Icc_of_Ioc hx))
```
in the places you currently write
```
integral_congr_Ioc (λ x hx, h x (mem_Icc_of_Ioc hx))
```